### PR TITLE
Adding field declaration for java language code generation

### PR DESCRIPTION
**Pull request by mattyb678**
on-behalf-of: @adobe <berteaux@adobe.com>

### CODE REVIEW

**Comment by snicoll** [comment]
Thanks for the PR @mattyb678 - We support Groovy and Kotlin as well and would need a support for these as well. Would you be ok to update the PR to support those as well?

**Comment by mattyb678** [comment]
@snicoll Not a problem, I can definitely do that. I'll tag you when I'm done and the PR is ready for review

**Comment by snicoll** [comment]
@mattyb678 do you have any update for us?

**Comment by mattyb678** [comment]
@snicoll Sorry, work has been keeping me really busy lately. I have this mostly done. I will finish it by this weekend.

**Comment by mattyb678** [comment]
@snicoll I think this PR is ready for a review. Although, the builds don't seem to have been triggered on the commits from yesterday or today.

File: initializr-generator/src/main/java/io/spring/initializr/generator/language/groovy/GroovyPrimitive.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*
+ * ************************************************************************
+ *  * ADOBE CONFIDENTIAL
```
[the remark this answers is not in the document]
Comment (mattyb678) [reply initializr-generator/src/main/java/io/spring/initializr/generator/language/groovy/GroovyPrimitive.java]: Sorry, I forgot to turn off the IDE from automatically adding my work one. Will fix this.

**Comment by mattyb678** [comment]
The code to do the field declaration is basically the same for all 3 languages. I added some base classes so that there could be a base SourceCodeWriter for all 3 languages. I wasn't aware that it was a conscious decision to duplicate the code. I'll go ahead and get rid of that and duplicate the code across the 3 languages. I'll ping you when I finish with that. 😄

**Comment by tcfly** [comment]
Sorry to bother,is this PR ready to merge? and when?thx.

**Comment by snicoll** [comment]
@tcfly in the comment above yours, @mattyb678 offered to review the proposal and ping us here when this is done. This hasn't happened yet. I forgot to add the `waiting-for-feedback` to make that more clear.

**Comment by mattyb678** [comment]
@tcfly I apologize, I have been busy with work. I will try to get this done in the next few days.

**Comment by mattyb678** [comment]
@snicoll This PR is ready for review. Thanks!

**Comment by tcfly** [comment]
@snicoll @mattyb678 Thanks for the updates.

**Comment by snicoll** [comment]
@mattyb678 well done and thank you very much for your contribution. I've decided to revert a fair amount of changes that I believe are not strictly related to field generation. I've also reverted the primitive support as we are using a more simpler/pragmatic approach for values at the moment.

I've also moved things around so that the diff only shows what was changed (the annotations handling code was moving at the bottom of the writer implementations which makes the diff bigger than it should be).

**Comment by mattyb678** [comment]
@snicoll Always happy to help! I really like your approach to the primitives and values 💯 Thanks for your time and work 😄

**Pull request closed without merging**